apps/src/applab/redux.js

```diff
@@ -0,0 +1,64 @@
+/** @file App Lab redux module */
+import _ from 'lodash';
+import { ApplabInterfaceMode } from './constants';
+import data from '../storage/redux/data';
+import screens from './redux/screens';
+import {reducers as jsDebuggerReducers} from '../lib/tools/jsdebugger/redux';
+import {
+  reducer as maker,
+  actions as makerActions,
+  selectors as makerSelectors,
+} from '../lib/kits/maker/redux';
+
+// Selectors
+
+export const selectors = {
+  // Curry maker selectors so they accept the root state.
+  maker: _.mapValues(makerSelectors, selector => state => selector(state.maker)),
```

Comment:
I like this, but I think this little piece of functionality should get moved out of here. Likewise, it should be **hard** to use `makerSelectors` that haven't been "mounted" to the right place in the state tree. Perhaps `lib/kits/maker/redux` should export a class that you have to instantiate with a root state selector. i.e something like this:

```jsx
// lib/kits/maker/redux.js
export class Selectors {
  constructor(getRoot) {
    this.getRoot = getRoot;
  }

  isEnabled(state) {
    return this.getRoot(state).enabled;
  }
}
```

Then:

```jsx
// applab/redux.js
import {Selectors as MakerSelectors} from '../lib/kits/maker/redux';
export const selectors = {
  maker: new MakerSelectors(state => state.maker),
};
```

Alternatively, perhaps we could do something like this:

```jsx
// lib/kits/maker/redux.js
class MountableSelectors {
  constructor(selectors) {
    this.selectors = selectors;
  }
  mount(getRoot) {
    return _.mapValues(this.selectors, selector => state => selector(getRoot(state)));
  }
}

export const selectors = new MountableSelectors({
  isEnabled: state => state.enabled;
});
```

Then:

```jsx
// applab/redux.js
import {selectors as makerSelectors} from '../lib/kits/maker/redux';
export const selectors = {
  maker: makerSelectors.mount(state => state.maker),
};
```

Ultimately, the _really_ slick way to do this sort of thing would be with [proxy objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy), which we unfortunately won't be able to do until we drop support for IE.

One final consideration here is in regards to the philosophical purpose of selectors, which is essentially to take a (deeply) nested state tree with potentially complex data structures, and flatten it into something that is easier to work with. So if you find yourself starting to "nest" selectors inside namespaces (in this case you have a `maker` selector namespace), particularly namespaces that exactly match the state tree hierarchy, then you might be making things more complicated rather than less complicated. I don't think that is necessarily happening here, but it's just something to consider.